zaza/openstack/charm_tests/nova/tests.py

```diff
@@ -470,10 +481,50 @@ def test_901_pause_resume(self):
         with self.pause_resume(['nova-compute']):
             logging.info("Testing pause resume")
 
+    def test_904_test_ceph_keys(self):
+        """Run pause and resume tests.
+
+        Pause service and check services are stopped then resume and check
+        they are started
+        """
+        # Expected default and alternate values
+        if zaza.model.get_application_config(
+                self.application_name)['libvirt-image-backend'].get(
+                    'value') != 'rbd':
+            return
+
+        regex = re.compile(r"^\[client.(.+)\]\n\tkey = (.+)$")
+        key_dict = {}
+
+        def check_keyring(app_name_dict_list):
+            for app_name_dict in app_name_dict_list:
+                for app_name, key_list in app_name_dict.items():
+                    for unit in zaza.model.get_units(
+                            app_name, model_name=self.model_name):
+                        for key_app_name in key_list:
+                            keyring_file = (
+                                '/etc/ceph/ceph.client.{}.keyring'.format(
+                                    key_app_name))
+                            data = str(generic_utils.get_file_contents(
+                                unit, keyring_file))
+
+                            result = regex.findall(data)
+                            self.assertEqual(2, len(result))
```

Comment:
nit: might be more obvious if you pull the key/value out here:

app_name, app_value = result

But just a nit

Reply:
It would raise value error if the length is different. The assert validation is less imprecise as an error message.